britnay268/OutTheGC
# Stretch 3

## Description
This update enhances the trip invitation functionality by adding a TripInvitation model with a status field and supporting CRUD operations. A new API endpoint was created to handle responses to invitations (approve or deny), with response messages refactored into structured response objects. The share trip endpoint now creates a tripInvitation entry in the database. Additional endpoints allow users to list and delete invitations, with improved error handling and logic consistency throughout the system. These changes streamline invitation management and improve the user experience.

## Related Issue
- #68 
- #69 
- #70 
- #71 
- #72

## Motivation and Context
These changes were necessary to enhance the functionality for managing trip invitations. The approval or denial of invitations should be clearer, with a response object providing structured information about the invitation status. Additionally, these changes improve the overall user experience by supporting CRUD operations for trip invitations, including creation, listing, responding, and deletion of invitations.

## How Can This Be Tested?
- Test the PATCH /trip/invitation/{invitationId} endpoint: Try sending requests with responses of approve and deny for a given trip invitation and verify that the correct status is updated in the database.

- Test the GET /trip/{userId}/invitations endpoint: Ensure the endpoint returns the list of invitations with correct data. Test out getting invitations without a status then with a status, for example 'pending'.

- Test the DELETE /trip/invitation/{invitationId} endpoint: Test deleting an invitation and ensure that the invitation is removed from the system.

- Test the POST /trip/invite endpoint: Ensure that a trip invitation entry is correctly created in the database when a trip is shared.


## Types of changes
